crates/bevy_transform/src/components/transform.rs

```diff
@@ -292,6 +292,29 @@ impl Transform {
         self.local_z()
     }
 
+    /// Translates this [`Transform`] within its local space using the provided translation vector.
+    ///
+    /// This method directly adjusts the [`Transform`]'s translation using the `translation` vector,
+    /// without considering the [`Transform`]'s current rotation. The translation happens in the
+    /// [`Transform`]'s local coordinate system, meaning the movement occurs along its local
+    /// axes (`x`, `y`, and `z`), regardless of the [`Transform`]'s orientation.
+    #[inline]
+    pub const fn translate(&mut self, translation: Vec3) {
+        self.translation += translation;
+    }
+
+    /// Translates this [`Transform`] within its local space, considering its rotation and orientation.
+    ///
+    /// This method translates this [`Transform`] based on its current rotation, so the movement
+    /// happens relative to the direction this [`Transform`] is facing. The `translation` vector is
+    /// transformed by the [`Transform`]'s rotation, allowing for natural directional movement.
+    /// For instance, moving "forward" means moving along the [`Transform`]'s forward direction
+    /// based on its current orientation, rather than just along the local z-axis.
+    #[inline]
+    pub const fn translate_with_local_rotation(&mut self, translation: Vec3) {
```

Comment:
This name is way too verbose to me, I would just call this `translate_local`, matching the existing `rotate_local`. Godot also has [`Transform3D.translated_local`](<https://docs.godotengine.org/en/stable/classes/class_transform3d.html#class-transform3d-method-translated-local>)

```suggestion
    pub const fn translate_local(&mut self, translation: Vec3) {
```

Reply:
I think my personal preference here would probably be having `translate` and `translate_local`, where the latter is with respect to the local frame, considering both rotation and scale. If you only want to consider rotation, then you can just do `transform.translate(transform.rotation * local_translation)`.